Freon combustion burn rate is now based on oxygen moles if there isn't enough oxygen for full burn. Fixes freon hotspots.

Pickle-Coding opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Closes #66783

For some reason the sign on the if check was reversed, which causes freon burn rate to be based on freon moles instead of oxygen moles if there is lacking oxygen for full burn. Not sure why it got flipped, but I'm just flipping the freon burn rates to correct it, so it should be based on oxygen moles if there isn't enough oxygen for fullburn.

Also for some reason freon fires checked if the holder was in a turf before reacting, and returning no reaction if it was not in a turf. This caused cold fire hotspots to kill themselves due to them not being a turf while making the gasmixture react, returning no gas reaction results then setting volume of fire to 0.

Fixes that by allowing freon to react in any gasmixture, but it can't create hot ice if it's not in a turf.
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
So freon fires don't instantly eat all of the oxygen anymore.
<!-- Argue for the merits of your changes and how they benefit the game, especially if they are controversial and/or far reaching. If you can't actually explain WHY what you are doing will improve the game, then it probably isn't good for the game in the first place. -->

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and it's effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl:
fix: Fixes freon fires being too efficient in oxygen deficient environments.
fix: Fixes freon fires hotspots not working properly.
add: Freon can now react in any gasmixture, instead of just turfs, but can't create hot ice if it's not in a turf.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Pickle-Coding committed:
Why?

For some reason the sign on the if check was reversed, which causes freon burn rate to be based on freon moles instead of oxygen moles if there is lacking oxygen for full burn.

Not sure why it got flipped, but I'm just flipping the freon burn rates to correct it, so it should be based on oxygen moles if there isn't enough oxygen for fullburn.

LemonInTheDark commented:
What pr flipped it?

Pickle-Coding commented:
Gas reaction cleanup one.

LemonInTheDark commented:
@TemporalOroboros was this intended? or just a clerical error

TemporalOroboros commented:
> @TemporalOroboros was this intended? or just a clerical error

Assuming that it wasn't that way before the cleanup PR, clerical error.

Pickle-Coding committed:
Fixes freon hotspots killing themselves.

For some weird reason, the reaction would return no reaction if the holder isn't a turf. This caused hotspots caused by freon fires to kill themselves because there was no reaction results after hotspot called react on a gasmixture that isn't in a turf.

This fixes that by removing the check at the start to check if the holder is a turf, and instead just check if it's a turf for hot ice spawning.

Pickle-Coding commented:
Why is map diffs renders running?

Pickle-Coding commented:
When map diff renders bot runs for more than 7 hours in a PR with no map changes.

tralezab commented:
just let the boy do his thing
